Images were named wrong and GITHUBURL env variable was not updated

betatim opened the pull request:
Solution: Skip repo url caching and always update env.

betatim committed:
Skip repo url caching and always update env

ibab commented on everware/spawner.py:
```diff
@@ -134,15 +134,13 @@ def options_from_form(self, formdata):
     def repo_url(self):
         return self.user_options['repo_url']
 
-    _escaped_repo_url = None
```
Maybe we can get rid of `_escaped_repo_url` completely?

ibab commented:
Had some problems verifying that the patch solves the problem, but after reinstall jupyterhub and the dockerspawner, it's working well for me.
I'm happy to merge this once you remove the `_escaped_repo_url`!

betatim commented:
How would we ditch it though? It is used to derive the image name :confused:

astiunov commented on everware/spawner.py:
```diff
@@ -134,15 +134,13 @@ def options_from_form(self, formdata):
+        repo_url = self.repo_url.translate(trans).lower()
+        if repo_url.endswith('.git'):
+            repo_url = repo_url[:-4]
+        self._escaped_repo_url = re.sub("_+", "_", repo_url)
```
We don't need self._escaped_repo_url

betatim commented:
This is buggy, I'll add a test and fix the bug.

betatim committed:
Fixup

betatim commented:
Does someone understand why this fails? On my machine it runs to completion. From the info in travis' log I can't really work it out :(

betatim commented:
Can reproduce this locally now. Started with setting up everware from scratch.

Problem: The container isn't passed a sensible URL to clone from

betatim commented:
Works with `dockerspawner` `master` but not the latest released version. Released version uses `self.env` and `master` uses `self.get_env()`

See jupyter/dockerspawner#71

Should we update our dependency to a commit to `master`? Or maybe pin it to a certain commit after the release.

betatim committed:
Move dockerspawner dependency past latest release

Need a fix that was applied after 0.2.0 was released,
so we now depend on a sepcific commit. As soon as there
is a new release we should go back to using that.

betatim commented:
- [ ] add a test that fails with the old code and passes after these changes
